// This proto file contains one service and three messages. 
// The service retrives the informations of the RAM, CPU, and Disk usage of a system. 


syntax="proto3"; 
import "google/protobuf/empty.proto";

// Defining the main service: SystemMonitor  
service SystemMonitor{
    // a streaming server rpc, response streaming rpc 
    rpc getRAMUsage (google.protobuf.Empty) returns ( stream RAMinfo) {}
    rpc getCPUUsage (google.protobuf.Empty) returns ( stream CPUFreq) {}
    rpc getDISkUsage (google.protobuf.Empty) returns ( stream DISkinfo) {}

} 
// The response message that contains the  informations about the RAM 
message RAMinfo {
    // The percentage of the total RAM storage   
    int32 total= 1; 
    // The percentage of the used RAM storage
    int32 used= 2; 
    // The percentage of the free RAM storage 
    int32 free= 3; 
}
// The response message that contains the frequency about the CPU
message CPUFreq {
    // The percentage of the current CPU frequency   
    int32 current= 1; 
    // The percentage of the min of the CPU frequency
    int32 min= 2; 
    // The percentage of the max of the CPU frequency 
    int32 max= 3; 
}
// The response message that contains the informations about the DISK 
 message DISkinfo {
    // The percentage of the total DISk storage   
    int32 total= 1; 
    // The percentage of the used DISk storage
    int32 used= 2; 
    // The percentage of the free DISk storage 
    int32 free= 3; 
}
